.github/setup_python/action.yml

```
@@ -1,27 +1,55 @@
 name: "Setup Python"
 description: "Sets up Python for use in actions with caching and copy localsettings"
 
+inputs:
+  requirements-file:
+    description: "if disabled this will only install runtime dependencies"
+    required: false
+    default: requirements-dev.txt
+
 runs:
   using: "composite"
   steps:
     - name: Setup python
       uses: actions/setup-python@v4
       with:
-        python-version: '3.10'
+        python-version: "3.10"
 
     - name: Restore venv cache
-      uses: syphar/restore-virtualenv@v1
       id: cache-virtualenv
+      uses: actions/cache@v3
+      with:
+        path: .venv
+        key: venv-${{ runner.os }}-${{ hashFiles( inputs.requirements-file ) }}-${{ inputs.requirements-file }}
+        restore-keys: venv-${{ runner.os }}-${{ hashFiles( inputs.requirements-file ) }}-${{ inputs.requirements-file }}
 
     - name: Restore pip download cache
-      uses: syphar/restore-pip-download-cache@v1
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ runner.os }}-${{ hashFiles( inputs.requirements-file ) }}-${{ inputs.requirements-file }}
+        restore-keys: pip-${{ runner.os }}-${{ hashFiles( inputs.requirements-file ) }}-${{ inputs.requirements-file }}
       if: steps.cache-virtualenv.outputs.cache-hit != 'true'
 
     - name: Install dependencies
-      run: pip install -r requirements-dev.txt
+      env:
+        requirements_txt: ${{ inputs.requirements-file }}
+      run: |
+        python -m venv .venv
+        source .venv/bin/activate
+        pip install -r $requirements_txt
```

Would the following also work?
```suggestion
      run: |
        python -m venv .venv
        source .venv/bin/activate
        pip install -r "${{ inputs.requirements-file }}"
```

---

seems to work: https://github.com/e-valuation/EvaP/actions/runs/8143458817/job/22255467262#step:3:61